sdk/core/typespec_client_core/src/base64.rs

```diff
@@ -213,14 +201,134 @@ where
 /// #[derive(Serialize)]
 /// struct SomeType {
 ///     #[serde(serialize_with = "base64::serialize_url_safe")]
-///     pub value: Option<Vec<u8>>,
+///     pub value: Vec<u8>,
 /// }
 /// ```
-pub fn serialize_url_safe<S, T>(to_serialize: &Option<T>, serializer: S) -> Result<S::Ok, S::Error>
+pub fn serialize_url_safe<S, T>(to_serialize: &T, serializer: S) -> Result<S::Ok, S::Error>
 where
     S: Serializer,
     T: AsRef<[u8]>,
 {
-    let encoded = to_serialize.as_ref().map(encode_url_safe);
-    <Option<String>>::serialize(&encoded, serializer)
+    let encoded = encode_url_safe(to_serialize.as_ref());
+    String::serialize(&encoded, serializer)
+}
```

Comment:
Nit: Tests?

Reply:
I'll add a few, but want to get this in so @jhendrixMSFT can use the merged commit for the emitter.